pkg/encryption/store/decrypt_store.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package store
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ethersphere/bee/pkg/encryption"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"golang.org/x/crypto/sha3"
+)
+
+type decryptingStore struct {
+	storage.Getter
+}
+
+func NewDecrypting(s storage.Getter) storage.Getter {
+	return &decryptingStore{s}
+}
+
+func (s *decryptingStore) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Address) (ch swarm.Chunk, err error) {
+	switch l := len(addr.Bytes()); l {
+	case swarm.HashSize:
+		// normal, unencrypted content
+		return s.Getter.Get(ctx, mode, addr)
+
+	case encryption.ReferenceSize:
+		// encrypted reference
+		ref := addr.Bytes()
+		address := swarm.NewAddress(ref[:swarm.HashSize])
+		ch, err := s.Getter.Get(ctx, mode, address)
+		if err != nil {
+			return nil, err
+		}
+
+		d, err := decryptChunkData(ch.Data(), ref[swarm.HashSize:])
+		if err != nil {
+			return nil, err
+		}
+		return swarm.NewChunk(address, d), nil
+
+	default:
+		return nil, storage.ErrReferenceLength
```

Comment:
you  see?this whole thing belongs to under `storage/encryption`  and encryption as used in pipeline should be a  putter, maybe lets sort this all out in a later PR?

Reply:
encryption cannot be a putter, since before you put the encrypted content you must hash it through bmt